dask/dataframe/core.py

```diff
@@ -1959,12 +1963,9 @@ def __dir__(self):
 
     def __getattr__(self, key):
         try:
-            return object.__getattribute__(self, key)
-        except AttributeError:
-            try:
-                return self[key]
-            except KeyError:
-                raise AttributeError()
```

Comment:
No. `__getattr__` is only called if the attribute isn't an explicit attribute of the object, so `object.__getattribute__(self, key)` would always raise an `AttributeError`, and the except path would always be taken.